django/contrib/contenttypes/fields.py

```diff
@@ -380,19 +384,19 @@ def _get_path_info_with_parent(self):
             path.extend(field.remote_field.get_path_info())
         return path
 
-    def get_path_info(self):
+    def get_path_info(self, filtered_relation=None):
         opts = self.remote_field.model._meta
         object_id_field = opts.get_field(self.object_id_field_name)
         if object_id_field.model != opts.model:
-            return self._get_path_info_with_parent()
+            return self._get_path_info_with_parent(filtered_relation)
         else:
             target = opts.pk
-            return [PathInfo(self.model._meta, opts, (target,), self.remote_field, True, False)]
+            return [PathInfo(self.model._meta, opts, (target,), self.remote_field, True, False, filtered_relation)]
 
-    def get_reverse_path_info(self):
+    def get_reverse_path_info(self, filtered_relation=None):
         opts = self.model._meta
         from_opts = self.remote_field.model._meta
-        return [PathInfo(from_opts, opts, (opts.pk,), self, not self.unique, False)]
+        return [PathInfo(from_opts, opts, (opts.pk,), self, not self.unique, False, filtered_relation)]
```

Comment:
Since you're editing it, using keyword args for all the `PathInfo` calls would make them clearer, like you added in `django/db/models/options.py`. We can always do this in a follow-up commit though.